File: backend/src/main/java/com/ioanrobertscumpu/backend/model/Produs.java
package com.ioanrobertscumpu.backend.model;

import java.util.UUID;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name = "produse")
public class Produs {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;
    private String nume;
    private String descriere;
    private String categorie;
    private String subcategorie;
    private String nume_vanzator;
    private double pret;
    private int cantitate;

    public Produs() {}

    public UUID getId() {
        return id;
    }

    public String getNume() {
        return nume;
    }

    public void setNume(String nume) {
        this.nume = nume;
    }

    public String getDescriere() {
        return descriere;
    }

    public void setDescriere(String descriere) {
        this.descriere = descriere;
    }

    public String getCategorie() {
        return categorie;
    }

    public void setCategorie(String categorie) {
        this.categorie = categorie;
    }

    public String getSubcategorie() {
        return subcategorie;
    }

    public void setSubcategorie(String subcategorie) {
        this.subcategorie = subcategorie;
    }

    public int getCantitate() {
        return cantitate;
    }

    public void setCantitate(int cantitate) {
        this.cantitate = cantitate;
    }

    @Override
    public String toString() {
        return nume + " " + descriere + " " + categorie + " " + subcategorie + " " + nume_vanzator + " " + pret + " " + cantitate + "\n";
    }
}

File: frontend/src/model/Produs.ts
class Produs {
    private id: string;
    private nume: string;
    private descriere: string;
    private categorie: string;
    private subcategorie: string;
    private numeVanzator: string;
    private pret: number;
    private cantitate: number;

    public constructor(
        id: string,
        nume: string,
        descriere: string,
        categorie: string,
        subcategorie: string,
        numeVanzator: string,
        pret: number,
        cantitate: number
    ) {
        this.id = id;
        this.nume = nume;
        this.descriere = descriere;
        this.categorie = categorie;
        this.subcategorie = subcategorie;
        this.numeVanzator = numeVanzator;
        this.pret = pret;
        this.cantitate = cantitate;
    }

    public getId(): string {
        return this.id;
    }

    public getNume(): string {
        return this.nume;
    }

    public getDescriere(): string {
        return this.descriere;
    }

    public getCategorie(): string {
        return this.categorie;
    }

    public getSubcategorie(): string {
        return this.subcategorie;
    }

    public getNumeVanzator(): string {
        return this.numeVanzator;
    }

    public getPret(): number {
        return this.pret;
    }

    public getCantitate(): number {
        return this.cantitate;
    }

    public toString(): string {
        return (
            this.nume +
            " " +
            this.descriere +
            " " +
            this.categorie +
            " " +
            this.subcategorie +
            " " +
            this.numeVanzator +
            " " +
            this.pret +
            " " +
            this.cantitate +
            "\n"
        );
    }
}

export default Produs;
